### PR TITLE
added Fuzzer to send udp commands

### PR DESCRIPTION
**Pull request by fred-he**

### CODE REVIEW

**Review by taylorlaekeman — CHANGES_REQUESTED** [review]
this code is based on an old version of the source code, I think it needs to be updated to fit with the system as it is now

File: src/Fozzie.java [hunk 1]
```diff
@@ -1,11 +1,94 @@
...
+    DatagramSocket clientSocket = new DatagramSocket();
+  	InetAddress IPAddress = InetAddress.getByName(DRONE_IP_ADDRESS);
+    
+    long startTime;
+    long endTime = 0;
+    
 
-  public static void main(String[] args) {
-    for (int i = 0; i < 100; i++)
-      System.out.println(i + " " + CommandFactory.makeCommand());
```
Comment (njjewers) [inline src/Fozzie.java]: Removing this line means that the command never regenerates, tight? You're just sending the same command over and over again

File: src/Fozzie.java [hunk 2]
```diff
@@ -1,11 +1,94 @@
...
+	private static final String DRONE_IP_ADDRESS = "192.168.1.1";
+  private static final String SAFE_MODE = "AT*PCMD=1,0,0,0,0,0\r"; // PCMD hovering mode
+
+  public static void main(String[] args) throws Exception {
+    try {
+  	  String cmd = args[0].toUpperCase();
+      int period = Int.parseInt(args[1]);
+      int valid = Int.parseInt(args[2]);
+
+      String commandString = CommandFactory.makeCommand(CommandFactory.CommandType.valueof(cmd), boolean valid!=0).toString();
```
Comment (njjewers) [inline src/Fozzie.java]: Move this block to the while loop so the command will be regenerated

File: src/Fozzie.java [hunk 3]
```diff
@@ -1,11 +1,94 @@
...
+import java.net.*;
+import java.lang.*;
 
 public class Fozzie {
+	
+	private static final int DRONE_PORT = 5556;
+	private static final int RECEIVE_PORT = 5554;
+	private static final int DRONE_NAVDATA_PORT = 5554;
+	private static final String DRONE_IP_ADDRESS = "192.168.1.1";
+  private static final String SAFE_MODE = "AT*PCMD=1,0,0,0,0,0\r"; // PCMD hovering mode
```
Comment (njjewers) [inline src/Fozzie.java]: IMO this should be "AT*REF=1,290717696\rAT*PCMD=1,0,0,0,0,0\r" - including the ref command is better parity with the app, and counteracts bad ref commands.

File: src/Fozzie.java [hunk 4]
```diff
@@ -1,11 +1,94 @@
...
+      int period = Int.parseInt(args[1]);
+      int valid = Int.parseInt(args[2]);
+
+      String commandString = CommandFactory.makeCommand(CommandFactory.CommandType.valueof(cmd), boolean valid!=0).toString();
+      if (valid) {
+        commandString += SAFE_MODE;
+      }
+    }
+
+    catch (Exception e) {
```
Comment (njjewers) [inline src/Fozzie.java]: There are only 3 args, IMO we should just print a usage message and die if there are no/bad args

File: src/Fozzie.java [hunk 5]
```diff
@@ -1,11 +1,94 @@
...
+      if (args.length != 0) {
+        System.out.println(e.getMessage());
+        System.out.println("Generate random command instead.");
+        int period = 50;
+      }
+      String commandString = CommandFactory.makeCommand();
+    }
+  	//setup serverSocket
+  	DatagramSocket serverSocket = new DatagramSocket(RECEIVE_PORT);
```
Comment (njjewers) [inline src/Fozzie.java]: This doesn't work, yeah? might as well remove it and all other code related to it

File: src/Fozzie.java [hunk 6]
```diff
@@ -1,11 +1,94 @@
...
+        String rcvd = "rcvd from " + dp.getAddress() + ", " + dp.getPort() + ": "+ new String(dp.getData(), 0, dp.getLength());
+        System.out.println(rcvd);
+        endTime = System.currentTimeMillis();
+        break;
+  		}
+  		catch (SocketTimeoutException e) {
+        System.out.println("Timeout reached!!! " + e);
+        break;
+  		}
+  	}
```
Comment (njjewers) [inline src/Fozzie.java]: Nix 69-84, seeing as they don't work

File: src/Fozzie.java [hunk 7]
```diff
@@ -1,11 +1,94 @@
...
+  	
+  	//Ensure for loop runs with a frequency of 1/PERIOD
+  	if(!((endTime - startTime) > period) && endTime != 0){
+  		Thread.sleep(period-(endTime-startTime));
+  	}
```
Comment (njjewers) [inline src/Fozzie.java]: likewise just replace this with a single thread.sleep

**Review by njjewers — CHANGES_REQUESTED** [review]
Did you build and test this?

File: src/Fuzzer.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
+package fozzie;
+
+import fozzie.fuzzer.CommandFactory;
+
+import java.io.*;
+import java.net.*;
+import java.lang.*;
+
+public class Fuzzer {
```
Comment (taylorlaekeman) [inline src/Fuzzer.java]: This is just a duplicate of Fozzie.java, right?  Can we delete this file?

Comment (fred-he) [reply src/Fuzzer.java]: yes

File: src/Fozzie.java (hunk 2 above)
Comment (fred-he) [reply src/Fozzie.java]: ? @njjewers

**Pull request merged**